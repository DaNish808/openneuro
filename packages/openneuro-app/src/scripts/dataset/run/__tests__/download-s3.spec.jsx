import React from 'react'
import { shallow } from 'enzyme'
import DownloadS3 from '../download-s3.jsx'

const datasetHash = 'dataset-hash'
const analysisId = 'analysis-id'

// Mock the needed configuration
jest.mock('../../../../../config', () => {
  return {
    aws: {
      s3: {
        analysisBucket: 'test-bucket',
      },
    },
  }
})

describe('dataset/run/DownloadS3', () => {
  it('should render something', () => {
    expect(
      shallow(<DownloadS3 datasetHash={datasetHash} analysisId={analysisId} />),
    ).toMatchSnapshot()
  })
  it('displays a help message if the s3:// URL is clicked', () => {
    const wrapper = shallow(
      <DownloadS3 datasetHash={datasetHash} analysisId={analysisId} />,
    )
    const s3Link = wrapper.find('a.s3-link')
    const preventDefault = jest.fn()
    s3Link.simulate('click', { preventDefault })
    expect(wrapper.find('Modal').props()['show']).toBe(true)
  })
  it('closes the modal when the close button is clicked', () => {
    const wrapper = shallow(
      <DownloadS3 datasetHash={datasetHash} analysisId={analysisId} />,
    )
    const s3Link = wrapper.find('a.s3-link')
    const preventDefault = jest.fn()
    s3Link.simulate('click', { preventDefault })
    // After opening the modal, hide it and see if that works correctly
    const modal = wrapper.find('Modal')
    modal.simulate('hide')
    expect(wrapper.find('Modal').props()['show']).toBe(false)
  })
})
