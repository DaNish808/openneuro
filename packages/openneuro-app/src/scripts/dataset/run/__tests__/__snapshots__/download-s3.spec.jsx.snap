// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/run/DownloadS3 should render something 1`] = `
<span>
  <a
    className="btn-warn-component s3-link"
    href="s3://test-bucket/dataset-hash/analysis-id"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-link"
    />
     S3 URL
  </a>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      >
        Download results from S3
      </ModalTitle>
      Results are available to download directly from S3.
    </ModalHeader>
    <hr
      className="modal-inner"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      Example download with
       
      <a
        href="http://docs.aws.amazon.com/cli/latest/userguide/installing.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        awscli
      </a>
      :
      <Well
        bsClass="well"
        bsSize="small"
      >
        aws --no-sign-request s3 sync 
        s3://test-bucket/dataset-hash/analysis-id
         
        <strong>
          destination-directory
        </strong>
      </Well>
    </ModalBody>
  </Modal>
</span>
`;
